internal/configs/virtualserver_test.go

```diff
@@ -1829,3 +1854,130 @@ func TestGenerateHealthCheck(t *testing.T) {
 		}
 	}
 }
+
+func TestGetEndpointsForVirtualServerEx(t *testing.T) {
+	name := "test-upstream"
```

Comment:
the name is used both for VirtualServer and the service, but not the upstream. That's why `test-upstream` might be confusing. Consider using a more general name like `test`.